vision_evaluation/evaluators.py

```diff
@@ -126,15 +126,20 @@ def calculate_score(self, average='macro'):
             by support (the number of true instances for each label).
         ``'samples'``:
             Calculate metrics for each instance, and find their average.
+        filter_out_zero_tgt : bool
+        Removes target columns that are all zero. For precision calculations this needs to be set to False, otherwise we could be removing FP
         """
+        filter_out_zero_tgt = kwargs.get('filter_out_zero_tgt', True)
         if self.all_predictions.size == 0:
             return 0.0
 
         tar_mat = _targets_to_mat(self.all_targets, self.all_predictions.shape[1])
         assert tar_mat.size == self.all_predictions.size
         result = 0.0
         if tar_mat.size > 0:
-            non_empty_idx = np.where(np.invert(np.all(tar_mat == 0, axis=0)))[0]
+            non_empty_idx = np.array([x for x in np.arange(tar_mat.shape[1])])
```

Comment:
`            non_empty_idx = np.where(np.invert(np.all(tar_mat == 0, axis=0)))[0] if filter_out_zero_tgt else np.array([x for x in np.arange(tar_mat.shape[1])])`